grammars/css.cson

```diff
@@ -329,183 +1273,348 @@
-          }
-          {
-            'include': '#string-double'
-          }
-          {
-            'match': '[^\'") \\t]+'
-            'name': 'variable.parameter.misc.css'
-          }
-        ]
+        'include': '#color-keywords'
       }
       {
-        'begin': '''(?x)
-          (?<![\\w-])
-          (
-            matrix(?:3d)?
-          | (?:translate|scale|rotate)(?:[XYZ]|3d)?
-          | skew[XY]?
-          | perspective
-          )
-          \\s*(\\()
-        '''
-        'beginCaptures':
-          '1':
-            'name': 'support.function.transform.css'
-          '2':
-            'name': 'punctuation.section.function.css'
-        'end': '\\)'
-        'endCaptures':
-          '0':
-            'name': 'punctuation.section.function.css'
-        'patterns': [
-          {
-            'include': '#numeric-values'
-          }
-          {
-            'match': ','
-            'name': 'punctuation.separator.css'
-          }
-        ]
+        'include': '#string'
       }
       {
-        'match': '\\!\\s*important'
+        'match': '!\\s*important(?![\\w-])'
         'name': 'keyword.other.important.css'
       }
     ]
+  'pseudo-classes':
+    'captures':
+      '1':
+        'name': 'punctuation.definition.entity.css'
+      '2':
+        'name': 'invalid.illegal.colon.css'
+    'match': '''(?xi)
+      (:)(:*)
+      (?: active|any-link|checked|default|disabled|empty|enabled|first
+        | (?:first|last|only)-(?:child|of-type)|focus|fullscreen|host|hover
+        | in-range|indeterminate|invalid|left|link|optional|out-of-range
+        | read-only|read-write|required|right|root|scope|target|unresolved
+        | valid|visited
+      )(?![\\w-])
+    '''
+    'name': 'entity.other.attribute-name.pseudo-class.css'
+  'pseudo-elements':
+    'captures':
+      '1':
+        'name': 'punctuation.definition.entity.css'
+      '2':
+        'name': 'punctuation.definition.entity.css'
+    'match': '''(?xi)
+      (?:
+        (::?)                       # Elements using both : and :: notation
+        (?: after
+          | before
+          | first-letter
+          | first-line
+          | (?:-(?:ah|apple|atsc|epub|hp|khtml|moz
+                  |ms|o|rim|ro|tc|wap|webkit|xv)
+              | (?:mso|prince))
+            -[a-z-]+
+        )
+        |
+        (::)                        # Double-colon only
+        (?: backdrop
+          | content
+          | grammar-error
+          | marker
+          | placeholder
+          | selection
+          | shadow
+          | spelling-error
+        )
+      )
+      (?![\\w-])
+    '''
+    'name': 'entity.other.attribute-name.pseudo-element.css'
   'rule-list':
-    'begin': '\\{'
+    'begin': '{'
     'beginCaptures':
       '0':
-        'name': 'punctuation.section.property-list.begin.css'
-    'end': '(\\})|(?<=\\})'
+        'name': 'punctuation.section.property-list.begin.bracket.curly.css'
+    'end': '(})|(?<=})'
```

Comment:
I'd issue a meaningful answer, but memories of my own code are buried under an avalanche of bikeshedding and whitespace picking. :p I've forgotten, so consider it fixed.